cooperator/tests/test_cooperator.py

```diff
@@ -52,96 +76,83 @@ def test_register_payment_for_capital_release(self):
-        self.assertEquals(self.request.partner_id.effective_date, Date.today())
+        self.assertEquals(
+            self.subscription_request_1.partner_id.effective_date, Date.today()
+        )
 
         share = partner.share_ids[0]
-        self.assertEquals(share.share_number, self.request.ordered_parts)
-        self.assertEquals(share.share_product_id, self.request.share_product_id)
+        self.assertEquals(share.share_number, self.subscription_request_1.ordered_parts)
+        self.assertEquals(
+            share.share_product_id, self.subscription_request_1.share_product_id
+        )
         self.assertEquals(share.effective_date, Date.today())
 
-    def test_user_rights(self):
-
-        request_values = {
-            "firstname": "test",
-            "lastname": "create request",
-            "email": "test@demo.net",
-            "address": "schaerbeekstraat",
-            "zip_code": "1111",
-            "city": "Brussels",
-            "country_id": self.ref("base.be"),
-            "date": Date.today(),
-            "source": "manual",
-            "ordered_parts": 3,
-            "share_product_id": self.browse_ref(
-                "cooperator.product_template_share_type_2_demo"
-            ).product_variant_id.id,
-            "lang": "en_US",
-        }
-
-        # test ir model access for base user
-        self.as_user()
-        read_request = self.browse_ref("cooperator.subscription_request_1_demo")
+    @users("demo")
+    def test_user_access_rules(self):
+        user_demo = self.env.ref("base.user_demo")
+        # class object was loaded with root user and its rights
+        resquest_as_user = self.subscription_request_1.sudo(user_demo)
         with self.assertRaises(AccessError):
-            read_request.name = "test write request"
+            resquest_as_user.name = "test write request"
         with self.assertRaises(AccessError):
-            self.env["subscription.request"].create(request_values)
+            self.env["subscription.request"].create(self.request_values)
         with self.assertRaises(AccessError):
-            read_request.unlink()
+            resquest_as_user.unlink()
 
-        share_line = self.browse_ref("cooperator.share_line_1_demo")
+        share_line_as_user = self.share_line.sudo(user_demo)
         with self.assertRaises(AccessError):
-            share_line.share_number = 3
-
-        # test ir model access for cooperator coop user
-        self.as_cooperator_user()
-        read_request = self.browse_ref("cooperator.subscription_request_1_demo")
-        read_request.name = "test write request"
-        create_request = self.env["subscription.request"].create(request_values)
+            share_line_as_user.share_number = 3
+
+    @users("user-cooperator")
+    def test_cooperator_access_rules(self):
+        cooperator_user = self.ref("cooperator.res_users_user_cooperator_demo")
+        resquest_as_cooperator = self.subscription_request_1.sudo(cooperator_user)
```

Comment:
are these `.sudo()` calls necessary with `@users()`?

Reply:
Yes, I left a comment about it in the `test_user_access_rules` function. I will _confer_ to it in other places.